botocore/handlers.py

```diff
@@ -619,11 +620,22 @@ def decode_list_object(parsed, context, **kwargs):
                 for member in parsed[top_key]:
                     member[child_key] = unquote_str(member[child_key])
 
+
+def convert_string_body_to_file_like_object(params, **kwargs):
+    if 'Body' in params and isinstance(params['Body'], six.string_types):
+
+        body = ensure_unicode(params['Body']).encode('utf-8')
```

Comment:
What if we just add an "ensure_bytes" function in compat?  The problem with `ensure_unicode` is that it requires the contents to be able to be decoded from utf-8, but s3 objects can have arbitrary binary data.  So instead, we'll have `ensure_bytes` be a noop if it's a bytes type, otherwise it will encode to utf-8 or whatever encoding you've provided.  That should handle the case of arbitrary binary data.

Side note:  if we don't have a test for this, it'd be worth adding.